src/shiver/views/corrections.py

```diff
@@ -0,0 +1,153 @@
+        # debye waller correction (disabled for now)
+        self.debye_waller_correction = QCheckBox("Debye-Waller correction")
+        self.debye_waller_correction.setEnabled(False)
+        # magentic structure factor (disabled for now)
+        self.magentic_structure_factor = QCheckBox("Magentic structure factor")
+        self.magentic_structure_factor.setEnabled(False)
+
+        # action group
+        # add a help button at the lower left corner
+        self.help_button = QPushButton("Help")
+        self.help_button.clicked.connect(self.help)
+        self.help_button.setToolTip("Open the help page")
+        self.help_button.setSizePolicy(QSizePolicy.Fixed, QSizePolicy.Fixed)
+        self.help_button.setFixedWidth(100)
+        self.help_button.setShortcut("F1")
+        # add a apply button
+        self.apply_button = QPushButton("Apply")
+        self.apply_button.clicked.connect(self.apply)
+        self.apply_button.setToolTip("Apply the corrections")
+        self.apply_button.setSizePolicy(QSizePolicy.Fixed, QSizePolicy.Fixed)
+        self.apply_button.setFixedWidth(100)
+        self.apply_button.setShortcut("Return")
+        # add a cancel button
+        self.cancel_button = QPushButton("Cancel")
+        self.cancel_button.clicked.connect(self.cancel)
+        self.cancel_button.setToolTip("Cancel the corrections")
+        self.cancel_button.setSizePolicy(QSizePolicy.Fixed, QSizePolicy.Fixed)
+        self.cancel_button.setFixedWidth(100)
+        self.cancel_button.setShortcut("Esc")
+        # config action group layout
+        action_layout = QHBoxLayout()
+        action_layout.addWidget(self.help_button)
+        action_layout.addSpacerItem(QSpacerItem(150, 0, QSizePolicy.Minimum, QSizePolicy.Expanding))
+        action_layout.addWidget(self.apply_button)
+        action_layout.addWidget(self.cancel_button)
+        action_layout.setAlignment(Qt.AlignLeft)
+
+        # config the layout
+        correction_layout = QVBoxLayout()
+        correction_layout.addLayout(detailed_balance_layout)
+        correction_layout.addWidget(self.hyspec_polarizer_transmission)
+        correction_layout.addWidget(self.debye_waller_correction)
+        correction_layout.addWidget(self.magentic_structure_factor)
+        correction_layout.addSpacerItem(QSpacerItem(0, 0, QSizePolicy.Minimum, QSizePolicy.Expanding))
+        correction_layout.addLayout(action_layout)
+
+        # set the layout
+        self.setLayout(correction_layout)
+
+    def help(self):
+        """Opens the help page"""
+        webbrowser.open("https://neutrons.github.io/Shiver/")
+
+    def apply(self):
+        """Apply the corrections"""
+        # NOTE: add return value to assist unit testing
+        logging.info("apply corrections")
+        output_ws = f"{self.ws_name}_correction"
+        # detailed balance
+        if self.detailed_balance.isChecked():
+            try:
+                ApplyDetailedBalanceMD(
```

Comment:
Anything interacting with the data in mantid should be happening in the model not the view.

Reply:
even for simple action like calling mantid algorithm?

Reply:
Yes

Reply:
I agree with Ross. The only two mentions of Mantid in the views are currently 1. importing a qt widget and 2. comments